core/src/main/kotlin/org/evomaster/core/problem/graphql/service/GraphQLFitness.kt

```diff
@@ -257,24 +258,114 @@ class GraphQLFitness : HttpWsFitness<GraphQLIndividual>() {
+
+                    var query = getQuery(returnGene, a)
+                    val bodyEntity = Entity.json("""
+            {"query" : "  { ${a.methodName}  ($printableInputGenes)  $query       } ","variables":null}
+        """.trimIndent())
+                    val invocation = builder.buildPost(bodyEntity)
+                    return invocation
+                }
+            } else {
+                if (returnGene.name.toLowerCase() == "scalar") {
+                    val bodyEntity = Entity.json("""
+            {"query" : "  { ${a.methodName}       } ","variables":null}
+        """.trimIndent())
+                    val invocation = builder.buildPost(bodyEntity)
+                    return invocation
+                } else {
+                    var query = getQuery(returnGene, a)
+                    val bodyEntity = Entity.json("""
+            {"query" : "  { ${a.methodName}  $query     } ","variables":null}
+        """.trimIndent())
+                    val invocation = builder.buildPost(bodyEntity)
+                    return invocation
+                }
+            }
+        } else if (a.methodType == GQMethodType.MUTATION) {
+            val printableInputGene: MutableList<String> = getPrintableInputGene(inputGenes)
+
+            var printableInputGenes = getPrintableInputGenes(printableInputGene)
 
-        val bodyEntity = Entity.json("""
-            {"query" : "$query","variables":null}
+            if (returnGene.name.toLowerCase() == "scalar") {
+                val bodyEntity = Entity.json("""
+            {"query" : " mutation{ ${a.methodName}  ($printableInputGenes)         } ","variables":null}
         """.trimIndent())
+                val invocation = builder.buildPost(bodyEntity)
+                return invocation
 
-        val invocation = builder.buildPost(bodyEntity)
+            } else {
+                var mutation = getMutation(returnGene, a)
+
+                val bodyEntity = Entity.json("""
+            { "query" : "mutation{    ${a.methodName}  ($printableInputGenes)    $mutation    }","variables":null}
+        """.trimIndent())
+                val invocation = builder.buildPost(bodyEntity)
+                return invocation
+            }
+        }
+        val invocation = builder.buildPost(Entity.json("""
+          """.trimIndent()))
         return invocation
     }
 
+    fun getMutation(returnGene: Gene, a: GraphQLAction): String {
+        var mutation = "{${returnGene.getValueAsPrintableString(mode = GeneUtils.EscapeMode.BOOLEAN_SELECTION_MODE)}}".replace("{${a.methodName}", "")
+        mutation = mutation.substring(0, mutation.length - 1)
+        return mutation
+    }
+
+    fun getQuery(returnGene: Gene, a: GraphQLAction): String {
+        var query = "{${returnGene.getValueAsPrintableString(mode = GeneUtils.EscapeMode.BOOLEAN_SELECTION_MODE)}}"
+        query = query.replace("{${a.methodName}", "", true)
+        query = query.substring(0, query.length - 1)
+        return query
```

Comment:
see previous comments

Reply:
.replace -> done.
.substring-> did not know how to simplify it